packages/headless/src/features/commerce/search/search-actions.ts

```diff
@@ -75,13 +89,17 @@ export const executeSearch = createAsyncThunk<
 
     return {
       response: fetched.success,
-      queryExecuted: state.commerceQuery?.query,
+      originalQuery:
```

Comment:
We are nearing the readability with those ternary and coalesce operator.

We should add a `querySelector` in `search-selectors.ts`.